matchms/Scores.py

```diff
@@ -128,39 +128,46 @@ def calculate(self) -> Scores:
                                                            is_symmetric=self.is_symmetric)
         return self
 
-    def scores_by_reference(self, reference: ReferencesType) -> numpy.ndarray:
+    def scores_by_reference(self, reference: ReferencesType,
+                            sort: bool = False) -> numpy.ndarray:
         """Return all scores (not sorted) for the given reference spectrum.
 
         Parameters
         ----------
         reference
             Single reference Spectrum.
+        sort
+            Set to True to obtain the scores in a sorted way (relying on the
+            sorting function from the given similarity_function).
         """
         assert reference in self.references, "Given input not found in references."
         selected_idx = int(numpy.where(self.references == reference)[0])
-        selected_scores = []
-        for i, score in enumerate(self._scores[selected_idx, :].copy()):
-            if isinstance(score, tuple):
-                selected_scores.append((self.queries[i], *score))
-            else:
-                selected_scores.append((self.queries[i], score))
+        if sort:
+            query_idx_sorted = self.similarity_function.sort(self._scores[selected_idx, :])
+            selected_scores = list(zip(self.queries[query_idx_sorted],
+                                       self._scores[selected_idx, query_idx_sorted].copy()))
+        else:
+            selected_scores = list(zip(self.queries, self._scores[selected_idx, :].copy()))
+
         return selected_scores
```

Comment:
Nice to see functional rewrite. Could be simplified to
```suggestion
            query_idx_sorted = self.similarity_function.sort(self._scores[selected_idx, :])
            return list(zip(self.queries[query_idx_sorted],
                                       self._scores[selected_idx, query_idx_sorted].copy()))
        return list(zip(self.queries, self._scores[selected_idx, :].copy()))
```

Reply:
Nice, thanks, looks much better this way (somehow I keep overlooking those things a bit...).